File: Inxi.NET/Hardware/System.vb

'    Inxi.NET  Copyright (C) 2020-2021  EoflaOE
'
'    This file is part of Inxi.NET
'
'    Inxi.NET is free software: you can redistribute it and/or modify
'    it under the terms of the GNU General Public License as published by
'    the Free Software Foundation, either version 3 of the License, or
'    (at your option) any later version.
'
'    Inxi.NET is distributed in the hope that it will be useful,
'    but WITHOUT ANY WARRANTY; without even the implied warranty of
'    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
'    GNU General Public License for more details.
'
'    You should have received a copy of the GNU General Public License
'    along with this program.  If not, see <https://www.gnu.org/licenses/>.

Public Class SystemInfo
    Inherits HardwareBase

    ''' <summary>
    ''' System name
    ''' </summary>
    Public Overrides ReadOnly Property Name As String
    ''' <summary>
    ''' Host name
    ''' </summary>
    Public ReadOnly Property Hostname As String
    ''' <summary>
    ''' Linux kernel version or Windows NT kernel version
    ''' </summary>
    Public ReadOnly Property SystemVersion As String
    ''' <summary>
    ''' System bits
    ''' </summary>
    Public ReadOnly Property SystemBits As Integer
    ''' <summary>
    ''' System name
    ''' </summary>
    Public ReadOnly Property SystemDistro As String
    ''' <summary>
    ''' Desktop manager
    ''' </summary>
    Public ReadOnly Property DesktopManager As String
    ''' <summary>
    ''' Window manager
    ''' </summary>
    Public ReadOnly Property WindowManager As String
    ''' <summary>
    ''' Display manager
    ''' </summary>
    Public ReadOnly Property DisplayManager As String

    ''' <summary>
    ''' Installs specified values parsed by Inxi to the class
    ''' </summary>
    Friend Sub New(Hostname As String, SystemVersion As String, SystemBits As Integer, SystemDistro As String, DesktopManager As String, WindowManager As String, DisplayManager As String)
        Me.Hostname = Hostname
        Me.SystemVersion = SystemVersion
        Me.SystemBits = SystemBits
        Me.SystemDistro = SystemDistro
        Name = SystemDistro
        Me.DesktopManager = DesktopManager
        Me.WindowManager = WindowManager
        Me.DisplayManager = DisplayManager
    End Sub

End Class

File: Inxi.NET/Hardware/System.cs
using InxiFrontend.Base;
using Newtonsoft.Json;

namespace InxiFrontend
{
    /// <summary>
    /// System information class
    /// </summary>
    public class SystemInfo : IHardware
    {
        [JsonProperty()]
        /// <summary>
        /// System name
        /// </summary>
        public string Name { get; set; }
        [JsonProperty()]
        /// <summary>
        /// Host name
        /// </summary>
        public string Hostname { get; private set; }
        [JsonProperty()]
        /// <summary>
        /// Linux kernel version or Windows NT kernel version
        /// </summary>
        public string SystemVersion { get; private set; }
        [JsonProperty()]
        /// <summary>
        /// System bits
        /// </summary>
        public int SystemBits { get; private set; }
        [JsonProperty()]
        /// <summary>
        /// System name
        /// </summary>
        public string SystemDistro { get; private set; }
        [JsonProperty()]
        /// <summary>
        /// Desktop manager
        /// </summary>
        public string DesktopManager { get; private set; }
        [JsonProperty()]
        /// <summary>
        /// Window manager
        /// </summary>
        public string WindowManager { get; private set; }
        [JsonProperty()]
        /// <summary>
        /// Display manager
        /// </summary>
        public string DisplayManager { get; private set; }

        /// <summary>
        /// Installs specified values parsed by Inxi to the class
        /// </summary>
        internal SystemInfo(string Hostname, string SystemVersion, int SystemBits, string SystemDistro, string DesktopManager, string WindowManager, string DisplayManager)
        {
            this.Hostname = Hostname;
            this.SystemVersion = SystemVersion;
            this.SystemBits = SystemBits;
            this.SystemDistro = SystemDistro;
            Name = SystemDistro;
            this.DesktopManager = DesktopManager;
            this.WindowManager = WindowManager;
            this.DisplayManager = DisplayManager;
        }
        [JsonConstructor()]
        public SystemInfo()
        {

        }
    }
}